.github/workflows/auto-label.yml

```diff
@@ -0,0 +1,41 @@
+name: Auto-label when user responds
+permissions:
+  issues: write
+
+on:
+  issue_comment:
+    types: [created]
+
+jobs:
+  run-check:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Add/Remove labels when user responds
+        uses: actions/github-script@v6
```

Comment:
I think you could do this more simply with the [`andymckay/labeler`](https://github.com/andymckay/labeler) action that we [used to use](https://github.com/microsoft/pyrx/commit/9d9b1974dca8bcd2da16334adb98bbceab9dd768) for `triage-needed`.

```yml
      - name: Add/Remove labels when user responds
        uses: andymckay/labeler@master
        if ${{ github.event.comment.sender == github.event.issue.user && contains(github.event.issue.labels.*.name, 'waiting for user response') && !contains(github.event.issue.labels.*.name, 'user responded') }}
        with:
          add-labels: 'user responded'
          remove-labels: 'waiting for user response'
```

I'm not sure if the `user` comparison will work. You might need to compare `user.id`.

Reply:
@debonte Thanks! I changed to `github.event.comment.user.login == github.event.issue.user.login` and tested it in my personal repo, and it worked.